internal/consensus/state.go

```diff
@@ -1382,7 +1382,7 @@ func (cs *State) defaultDoPrevote(height int64, round int32) {
 		return
 	}
 
-	if cs.Proposal.POLRound == -1 && cs.LockedRound == -1 && !cs.proposalIsTimely() {
+	if cs.isPBTSEnabled(height) && cs.Proposal.POLRound == -1 && cs.LockedRound == -1 && !cs.proposalIsTimely() {
```

Comment:
Moreover, I would be more didactic with the code reader here.

If POLRound == 1 and PBTS is enabled. Then inside the if check timely. If it fails, log this and prevote nil. But this is only a suggestion.

Reply:
> I know it was not introduced here, but why we have the third condition (`cs.LockedRound == -1`)?

If `cs.LockedRound == -1` then there is a chance that the block will be accepted in line 1471, since its timeliness has been tested and passed before, speeding up a decision.

Reply:
If `cs.Proposal.POLRound == -1` we don't even get there.